internal/config/load_test.go

```diff
@@ -614,12 +614,16 @@ pre-push:
 		}
 
 		t.Run(fmt.Sprintf("%d: %s", i, tt.name), func(t *testing.T) {
-			if err := fs.WriteFile(filepath.Join(root, "lefthook.yml"), []byte(tt.global), 0o644); err != nil {
-				t.Errorf("unexpected error: %s", err)
+			if tt.global != "" {
```

Comment:
Only write to the config file if we explicitly specify its content. That way, if `.lefthook.yml` is written to, an empty `lefthook.yml` doesn't take precedence